Source/ARTRealtime.m

```diff
@@ -849,28 +849,22 @@ - (void)onHeartbeat {
 - (void)onConnected:(ARTProtocolMessage *)message {
     _renewingToken = false;
     
-    // Resuming
```

Comment:
I think because this makes sense - you don't want `[channel reattachWithMetadata:metadata]` be called in any other connection state, except `CONNECTING` upon receive `CONNECTED` protocol message. I don't know why previous implementation was for all states.

Reply:
I agree it makes sense, perhaps would have been better (for the sake of somebody who sees this commit in the future) in a separate commit with an explanation. Will leave it to you whether you want to do that (again, if so would probably be best to do the rebase once ready to merge).